client/packet/noop_reporter.go

```diff
@@ -0,0 +1,60 @@
+package packet
+
+import (
+	"context"
+	"io"
+
+	"github.com/packethost/pkg/log"
+	"github.com/tinkerbell/boots/client"
+)
+
+type noOpReporter struct {
+	logger log.Logger
+}
+
+func (c *noOpReporter) PostHardwareComponent(ctx context.Context, hardwareID client.HardwareID, body io.Reader) (*client.ComponentsResponse, error) {
+	return nil, nil
+}
+
+func (c *noOpReporter) PostHardwareEvent(ctx context.Context, id string, body io.Reader) (string, error) {
+	return "", nil
+}
+
+func (c *noOpReporter) PostHardwarePhoneHome(ctx context.Context, id string) error {
+	return nil
+}
+
+func (c *noOpReporter) PostHardwareFail(ctx context.Context, id string, body io.Reader) error {
+	return nil
+}
+
+func (c *noOpReporter) PostHardwareProblem(ctx context.Context, id client.HardwareID, body io.Reader) (string, error) {
+	return "", nil
+}
+
+func (c *noOpReporter) PostInstancePhoneHome(context.Context, string) error {
+	return nil
+}
+
+func (c *noOpReporter) PostInstanceEvent(ctx context.Context, id string, body io.Reader) (string, error) {
+	return "", nil
+}
+
+func (c *noOpReporter) PostInstanceFail(ctx context.Context, id string, body io.Reader) error {
+	return nil
+}
+
+func (c *noOpReporter) PostInstancePassword(ctx context.Context, id, pass string) error {
+	return nil
+}
+
+func (c *noOpReporter) UpdateInstance(ctx context.Context, id string, body io.Reader) error {
+	return nil
+}
+
+// NewNoOpReporter returns a reporter that does nothing.
```

Comment:
yea, thanks

Reply:
Isn't that just "Where its useful". We could mentioned tests but do code docs really need to list where to use something as open as this?

Reply:
I forgot to add this to `cmd/main.go` for non-cacher backends. Rather than refactor every instance where reporting is touched, I thought it might be easier to have a no-op and later refactor that

Reply:
@chrisdoherty4 normally yeah, but documenting when/why someone would want one seems most useful for noop vs EM, or tink or w/e else because its kind of obvious from the name (but being explicit in the documentation doesn't hurt for those cases either).